junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/ForkJoinPoolHierarchicalTestExecutorService.java

```diff
@@ -155,29 +157,34 @@ public void invokeAll(List<? extends TestTask> tasks) {
 			new ExclusiveTask(tasks.get(0)).execSync();
 			return;
 		}
-		Deque<ExclusiveTask> nonConcurrentTasks = new LinkedList<>();
+		Deque<ExclusiveTask> isolatedTasks = new LinkedList<>();
+		Deque<ExclusiveTask> sameThreadTasks = new LinkedList<>();
```

Comment:
🤔 why are we using `LinkedList` instead of `ArrayDeque`?
It will probably not really matter performance wise.

Reply:
I was wondering that, too, but kept using `LinkedList` since it was already in place.